tests/losses/test_edge_crossings_loss.py

```diff
@@ -38,7 +39,7 @@ def test_find_nearest_triangles(loss_fn):
 
     nearest = loss_fn.find_nearest_triangles(vertices, faces)
     assert nearest.shape[0] == faces.shape[0]
-    assert nearest.shape[1] == 1  # k-1 = 1, since k=2
+    assert nearest.shape[1] == k_val
```

Comment:
I believe the comment from before does that fit anymore. The function will only return less than `k` if the number of centroids are less than `k`.